### ISSUE TITLE
Information about DeviceMessageLockLostException

### ISSUE THREAD

**Issue by mathisharp** [Azure/azure-iot-sdk-csharp#90]
Hello,
I have issue with deviceMessagelostlockexception. I need to understand when this exception will be raised and also whether the connection will reestablish by itself and about the data send during that time will be rejected or will the abandoned message will be re-en-queued. My scenario is like sending 200k tags @ 10s collection rate 20k per second. So during this run we get devicelostlockexception. My concern is whenever this error raises whether the failed message is being re-enqueued or not.

**Comment by jasmineymlo** [comment]
@mathisharp, would you clarify if you are sending device-to-cloud or cloud-to-device messages?  Thanks.

**Comment by mathisharp** [comment]
@jasmineymlo , I am sending device-to-cloud message. Using Amqp_WebSocket_Only protocol.
Thanks.

**Comment by jasmineymlo** [comment]
@mathisharp AFAIK, the DeviceMessageLockLostException would show up on the device client when trying to complete message received from iothub which it was not able to find the lock.  For your reference you can see the Message Lifecyle https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messaging#cloud-to-device-messages.

We had seen this happens if multiple devices are using the same device ID.  You might want to check on that.  Thanks.

**Comment by mathisharp** [comment]
We are using the device Id only once at a time. We repeatedly face this issue when the completeasync action in performed in disorder way. I have attached the excel for the message flow between receiveasync and completeasync. Please let us know if you need anymore information.
[MessageFlowSheet.xlsx](https://github.com/Azure/azure-iot-sdk-csharp/files/1033454/MessageFlowSheet.xlsx)

**Comment by mtuchkov** [comment]
@mathisharp What is the exception rate?

**Comment by mathisharp** [comment]
@mtuchkov I have created new ticket with the sample to reproduce the issue. Please refer this link - https://github.com/Azure/azure-iot-sdk-csharp/issues/117

**Comment by tameraw** [comment]
Linking to #117 and closing this one.
